package {{packageName}}.error.domain;

import static org.assertj.core.api.Assertions.*;

import org.junit.jupiter.api.Test;
import {{packageName}}.UnitTest;

@UnitTest
class AccountExceptionTest {

  @Test
  void shouldGetAccountException() {
    AccountException exception = new AccountException();
    assertThat(exception.getMessage()).isNull();
  }

  @Test
  void shouldAccountExceptionWithMessage() {
    AccountException exception = new AccountException("Hello JHipster");
    assertThat(exception.getMessage()).isEqualTo("Hello JHipster");
  }

  @Test
  void shouldAccountExceptionWithCause() {
    NullPointerException nullPointerException = new NullPointerException();
    AccountException exception = new AccountException("Hello JHipster", nullPointerException);

    assertThat(exception.getMessage()).isEqualTo("Hello JHipster");
    assertThat(exception.getCause()).isInstanceOf(NullPointerException.class);
  }
}
